R/zchunk_L114.bcoc_en_R_S_T_Y.R

```diff
@@ -38,43 +41,112 @@ module_emissions_L114.bcoc_en_R_S_T_Y_DISABLED <- function(command, ...) {
+      # the appropriate function to use whenever this table is joined in is the 'first' match. Dealing with this
+      # would require a third "by" column (fuel), which would require modifications to the GCAM_sector_tech
+      # mapping table.
+      left_join_keep_first_only(select(GCAM_sector_tech, sector, technology,BCOC_agg_sector, BCOC_agg_fuel),
+                                by = c("sector", "technology"))
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Prepare the BC/OC default emissions factors (computed in L104) for joining into the BCOC driver table
+    BCOC_default_coefs <- L104.bcoc_tgej_USA_en_T_1990 %>%
+      gather(Non.CO2, emissions.factor, -sector, -technology) %>%
+      # The Non.CO2 column has "bc_em_factor" where we want "BC", and so on for OC
+      mutate(Non.CO2 = toupper(substr(Non.CO2, 1, 2))) %>%
+      # The "sector" and "technology" columns should be renamed to "BCOC_agg_sector" and "BCOC_agg_fuel"
+      rename(BCOC_agg_sector = sector, BCOC_agg_fuel = technology)
+
+    # Join in the coefficients and multiply to get the unscaled BC and OC emissions by
+    # region, sector, and technology for the year 2000
+    BCOC_unscaled_emissions <- BCOC_drivers %>%
+      # inner join because there are NA's and they are intended to be dropped
+      inner_join(BCOC_default_coefs, by = c("BCOC_agg_sector", "BCOC_agg_fuel", "Non.CO2")) %>%
+      mutate(unscaled_emissions = energy * emissions.factor) %>%
+      # Match in the RCP sectors used for scaling these emissions prior to aggregation
+      left_join_keep_first_only(select(GCAM_sector_tech, sector, technology, RCP_agg_sector),
+                                 by = c("sector", "technology"))
+
+    # Aggregate by the broader "RCP" categories for computing scalers
+    BCOC_unscaled_emissions_RCP <- BCOC_unscaled_emissions %>%
+      group_by(GCAM_region_ID, Non.CO2, RCP_agg_sector) %>%
+      summarise(unscaled_emissions = sum(unscaled_emissions)) %>%
+      ungroup()
+
+    # Prepare the RCP inventory estimates for joining into the table of unscaled emissions by RCP inventory categories
+    RCP_BC_2000$Non.CO2 <- "BC"
+    RCP_OC_2000$Non.CO2 <- "OC"
+    BCOC_emissions_RCP <- bind_rows(RCP_BC_2000, RCP_OC_2000) %>%
+      gather(RCP_agg_sector, scaled_emissions, -Country, -iso, -Non.CO2) %>%
+      mutate(scaled_emissions = scaled_emissions * CONV_KG_TO_TG)
+    # in the old data system, the Bahamas had a missing value for the iso, so their emissions data were dropped
```

Comment:
We only want to drop the Bahamas so that the oldnew comparison check works. The RCP_BC_2000 and RCP_OC_2000 files have been modified from the old data system, in that the "#N/A" for the iso of The Bahamas has been replaced with "bhs".